src/range-to-string.js

```diff
@@ -0,0 +1,68 @@
+/* global Node */
+
+/**
+ * Return the next node after `node` in a tree order traversal of the document.
+ */
+function nextNode(node, skipChildren) {
+  if (!skipChildren && node.firstChild) {
+    return node.firstChild
+  } else if (node.nextSibling) {
+    return node.nextSibling
+  }
+  while (node.parentNode) {
+    node = node.parentNode
+    if (node.nextSibling) {
+      return node.nextSibling
+    }
+  }
+  /* istanbul ignore next */
+  return null
+}
+
+function firstNode(range) {
+  if (range.startContainer.nodeType === Node.ELEMENT_NODE) {
+    const node = range.startContainer.childNodes[range.startOffset]
+    return node || range.startContainer
```

Comment:
Should this be `node || nextNode(range.startContainer)`? If the offset is past all the children, that would logically place the contents of the range after the start container, to me. Of course, this is an exceptional case that I'm not even sure is possible with browser-supplied range support. Your thoughts? I'm not committed either way.